qelem/lib/infrastructure/answer/answer_repoistory.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:qelem/domain/answer/answer.dart';
+import 'package:qelem/infrastructure/answer/answer_api.dart';
+import 'package:qelem/infrastructure/answer/answer_dto.dart';
+import 'package:qelem/infrastructure/answer/answer_form_dto.dart';
+import 'package:qelem/infrastructure/answer/answer_model_mapper.dart';
+import 'package:qelem/infrastructure/common/qelem_http_exception.dart';
+import 'package:qelem/util/either.dart';
+import 'package:qelem/util/error.dart';
+
+class AnswerRepository {
+  final AnswerApi answerApi;
+  AnswerRepository(this.answerApi);
+
+  Future<Either<Answer>> createAnswer(
+      {required AnswerFormDto answerFormDto}) async {
+    try {
+      AnswerDto answer = await answerApi.createAnswer(answerFormDto);
+      return Either(val: answer.toAnswer());
+    } on QHttpException catch (exception) {
+      return Either(error: Error(exception.message));
+    }
+  }
+
+  Future<void> deleteAnswer(int answerId) async {
+    try {
+      await answerApi.deleteAnswer(answerId);
+    } on QHttpException catch (exception) {
+      throw Either(error: Error(exception.message));
+    }
+  }
+
+  Future<Either<AnswerDto>> updateAnswer(
+      AnswerFormDto answerFormDto, int answerId, String content) async {
+    try {
+      AnswerDto updatedAnswer =
+          await answerApi.updateAnswer(answerFormDto, answerId, content);
+      return Either(val: updatedAnswer);
+    } on QHttpException catch (exception) {
+      return Either(error: Error(exception.message));
+    }
+  }
+
+  Future<void> upvoteAnswer(int answerId) async {
+    try {
+      await answerApi.upvoteAnswer(answerId);
+    } catch (exception) {
+      rethrow;
+    }
+  }
```

Comment:
How would the UI update when there is an upvote if there is not response (void response) from the backend?

Reply:
what if i send .getAnswerById to the api and return the response?

Reply:
see my pull request [question-repository](https://github.com/s9uidward/Qelem/pull/67/files#diff-2046dd6c6e9f333e77f26daaecb03c880a75b815a86bd59f4b5b2b9d361fe72a). There is a huge similarity between this and the one I did. If you have other suggestion, do let me know as it would also help me.